pandas/core/arrays/_mixins.py

```diff
@@ -68,7 +68,7 @@ def _validate_fill_value(self, fill_value):
     # TODO: make this a cache_readonly; for that to work we need to remove
     #  the _index_data kludge in libreduction
     @property
-    def shape(self) -> Tuple[int, ...]:
+    def shape(self) -> Tuple[int]:
```

Comment:
datetimearray, timedeltaarray can be 2d sometimes

Reply:
is that accounted for in the changes further down to pandas/core/internals/blocks.py

Reply:
blocks shouldnt come in to it.  Part of the idea of the mixin in this file is to support 2D EAs.

Reply:
the backing data could be 2d? but isn't the EA only ever 1d?

Reply:
There are corner cases where a DatetimeArray/TimedeltaArray can be 2D in ops.  (and if i get my way, eventually we'll support 2D in general)

Reply:
ok i'll revert some of these changes for now. the base array class will need to allow a 2 tuple then. rather than Tuple[int, ...] we may want to go with Union[Tuple[int] ,Tuple[int, int] ] to only allow 1d and 2d

Reply:
might be worth having an alias: Shape1D, Shape2D or similar

Reply:
ended up going down a rabbit hole on this. so will come back another day refreshed.